### ISSUE TITLE
Confusing aria-label for first and last dates in selected of Range Calendar

### ISSUE THREAD

**Issue by xtstoll** [adobe/react-spectrum#6043]
### Provide your feedback here.

When using a screen reader to select a date range in the [Range Calendar](https://react-spectrum.adobe.com/react-spectrum/RangeCalendar.html), the `aria-label` of the first and last dates updates when the information about the selection.

However, this creates an `aria-label` value that is very long, and in most cases, very confusing to understand.

### 🔦 Context

Before selecting anything this is the `aria-label` of the first date I will select:
`aria-label="Tuesday, March 12, 2024"`

After selecting the range, it updates like this:
`aria-label="Selected Range: Tuesday, March 12 to Saturday, March 16, 2024, Tuesday, March 12, 2024 selected"`

This is confusing to hear three dates at once. Perhaps, it will be less confusing to hear like this:
`aria-label="Range: March 12 to March 16, Date: Tuesday, March 12, 2024 selected"`

Here is a video of NVDA 2023.3.4 should this:
https://github.com/adobe/react-spectrum/assets/24814666/706c492e-45c9-41b7-a0ba-5e1a3669580e

### Version

3.34.0

### What browsers are you seeing the problem on?

Chrome, Safari

### What operating system are you using?

Windows

**Comment by LFDanLu** [comment]
Thank you for the feedback. Personally I'm a bit torn on the balance here between being informational versus going for brevity. It feels like the specific week day information (and year since that will also be announced if the range spans across multiple years) is important information, especially if the user wasn't the one who set that range in the first place (i.e. it was pre-set by the app). However, I guess it depends based on the informational context the calendar resides in and the limits set on the RangeCalendar itself (aka if the RangeCalendar only spans a couple weeks or a month then the some of the information can be inferred).

cc @majornista any opinions here?

**Comment by majornista** [comment]
I agree with @LFDanLu, that including detail of a day of the week, and/or year when the range spans multiple years, seems like important information. We want to ensure that complete selected range information gets announced when a range selection is completed, and when a user navigating the grid enters a selected range. We could try to include information on the selected range using `aria-describedby`, but announcement of `aria-describedby` may be less reliable than we would need it to be.
